modules/medianetAnalyticsAdapter.js

```diff
@@ -382,46 +435,68 @@ class Auction {
   }
 }
 
-function auctionInitHandler({auctionId, timestamp, bidderRequests}) {
+function auctionInitHandler({auctionId, adUnits, timeout, timestamp, bidderRequests}) {
   if (auctionId && auctions[auctionId] === undefined) {
     auctions[auctionId] = new Auction(auctionId);
     auctions[auctionId].auctionInitTime = timestamp;
   }
+  addAddSlots(auctionId, adUnits, timeout);
   const floorData = utils.deepAccess(bidderRequests, '0.bids.0.floorData');
   if (floorData) {
     auctions[auctionId].floorData = {...floorData};
   }
 }
 
-function bidRequestedHandler({ auctionId, auctionStart, bids, start, timeout, uspConsent, gdpr }) {
+function addAddSlots(auctionId, adUnits, tmax) {
+  adUnits = adUnits || [];
+  const groupedAdUnits = utils.groupBy(adUnits, 'code');
+  Object.keys(groupedAdUnits).forEach((adUnitCode) => {
+    const adUnits = groupedAdUnits[adUnitCode];
+    const supplyAdCode = utils.deepAccess(adUnits, '0.adUnitCode') || adUnitCode;
+    let context = '';
+    let adext = {};
+
+    const mediaTypeMap = {};
+    const oSizes = {banner: [], video: []};
+    adUnits.forEach(({mediaTypes, sizes, ext}) => {
+      mediaTypes = mediaTypes || {};
+      adext = Object.assign(adext, ext || utils.deepAccess(mediaTypes, 'banner.ext'));
+      context = utils.deepAccess(mediaTypes, 'video.context') || '';
```

Comment:
Have fixed it!

```js
context = utils.deepAccess(mediaTypes, 'video.context') || context;
```

In case of twin ad units, there can only be a single valid `context` value, thus if we get a valid `context` value in the first iteration and in second iteration `undefined` or `''`, it will default to `context` value from the first iteration. 

Thanks for point it out :) !